src/Auth/Eloquent/User.php

```diff
@@ -323,7 +323,7 @@ public function lastLogin()
             return null;
         }
 
-        return Carbon::createFromFormat($this->model()->getDateFormat(), $date);
+        return $date instanceof Carbon ? $date->format($this->model()->getDateFormat()) : Carbon::createFromFormat($this->model()->getDateFormat(), $date);
```

Comment:
This isnt returning the Carbon instance when date is Carbon... it will return a string (eg 2022-02-02 01:01:01)... so if the date is Carbon you get a different type returned to when its not.

Surely it should be 

`return $date instanceof Carbon ? $date : Carbon::createFromFormat($this->model()->getDateFormat(), $date);`

Reply:
Yep you are right. Pushed a fix for that. Thank you!